sql/tests: TestRandomSyntaxGeneration failed [ALTER CHANGEFEED with wrong type can cause node to crash]

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11261535?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11261535?buildTab=artifacts#/) on master @ [a598a75435f6426236dde569191fbd95871f24ae](https://github.com/cockroachdb/cockroach/commits/a598a75435f6426236dde569191fbd95871f24ae):


```
=== RUN   TestRandomSyntaxGeneration
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration213973572
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:98: 
        Test server was configured to route SQL queries to a secondary tenant (virtual cluster).
        If you are only seeing a test failure when this message appears, there may be a problem
        specific to cluster virtualization or multi-tenancy.
        
        To investigate, consider using "COCKROACH_TEST_TENANT=true" to force-enable just
        the secondary tenant in all runs (or, alternatively, "false" to force-disable), or use
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
    test_server_shim.go:162: 
        Test server was configured to route SQL queries to a secondary tenant (virtual cluster).
        If you are only seeing a test failure when this message appears, there may be a problem
        specific to cluster virtualization or multi-tenancy.
        
        To investigate, consider using "COCKROACH_TEST_TENANT=true" to force-enable just
        the secondary tenant in all runs (or, alternatively, "false" to force-disable), or use
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
    rsg_test.go:844: 5s of 5m0s: 7523 executions, 477 successful
    rsg_test.go:844: 10s of 5m0s: 13907 executions, 858 successful
    rsg_test.go:844: 15s of 5m0s: 17805 executions, 1098 successful
    rsg_test.go:844: 20s of 5m0s: 21842 executions, 1326 successful
    rsg_test.go:844: 25s of 5m0s: 25574 executions, 1557 successful
    rsg_test.go:844: 30s of 5m0s: 29268 executions, 1764 successful
    rsg_test.go:844: 35s of 5m0s: 32804 executions, 1954 successful
    rsg_test.go:844: 40s of 5m0s: 36283 executions, 2163 successful
    rsg_test.go:844: 45s of 5m0s: 39891 executions, 2391 successful
    rsg_test.go:844: 50s of 5m0s: 43724 executions, 2588 successful
    rsg_test.go:844: 55s of 5m0s: 47221 executions, 2776 successful
    rsg_test.go:844: 1m0s of 5m0s: 50925 executions, 3012 successful
    rsg_test.go:844: 1m5s of 5m0s: 54216 executions, 3198 successful
    rsg_test.go:844: 1m10s of 5m0s: 57722 executions, 3416 successful
    rsg_test.go:844: 1m15s of 5m0s: 61579 executions, 3624 successful
    rsg_test.go:844: 1m20s of 5m0s: 64720 executions, 3832 successful
panic: expected *DInt, found *tree.DBitArray

goroutine 429495 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDInt({0x7a4e870?, 0xc00c28f480})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:723 +0x8e
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.alterChangefeedPlanHook.func1({0x7a30500, 0xc00e478e70}, {0xc00730fef8?, 0xc00730ffd0?, 0x0?}, 0xc00d210300)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/alter_changefeed_stmt.go:91 +0xd2
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1()
	github.com/cockroachdb/cockroach/pkg/sql/planhook.go:206 +0xa4
created by github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec
	github.com/cockroachdb/cockroach/pkg/sql/planhook.go:204 +0x1b2
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #108233 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30495

trivial reproduction, which causes a node to crash:
```
> alter changefeed '0101'::varbit add b;

panic: expected *DInt, found *tree.DBitArray

goroutine 2827 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDInt({0x108ced1c0?, 0x14006207100})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:723 +0x80
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.alterChangefeedPlanHook.func1({0x108cceec0, 0x14005b33ef0}, {0x0?, 0x0?, 0x0?}, 0x14004fde360)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/alter_changefeed_stmt.go:91 +0xa8
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1()
	github.com/cockroachdb/cockroach/pkg/sql/planhook.go:206 +0x88
created by github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec
	github.com/cockroachdb/cockroach/pkg/sql/planhook.go:204 +0x1b8
````

--- a/pkg/ccl/changefeedccl/alter_changefeed_stmt.go
+++ b/pkg/ccl/changefeedccl/alter_changefeed_stmt.go
@@ -84,7 +84,7 @@ func alterChangefeedPlanHook(
 	}
 
 	fn := func(ctx context.Context, _ []sql.PlanNode, resultsCh chan<- tree.Datums) error {
-		typedExpr, err := alterChangefeedStmt.Jobs.TypeCheck(ctx, p.SemaCtx(), types.Int)
+		typedExpr, err := tree.TypeCheckAndRequire(ctx, alterChangefeedStmt.Jobs, p.SemaCtx(), types.Int, "get-job-id")
 		if err != nil {
 			return err
 		}

